ide/commons-gwt/src/main/java/org/eclipse/che/api/promises/client/BiOperation.java

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.promises.client;
+
+/**
+ * Interface for an 'operation', as a binary function without a return value,
+ * only side-effects, but without the burden of having a callback with <code>
+ * Void</code> parameter.
+ *
+ * @param <A1>
+ *         the type of the first argument
+ * @param <A2>
+ *         the type of the second argument
+ */
+public interface BiOperation<A1, A2> {
```

Comment:
Actually it is a good idea though I don't think it is necessary right now because we're in a process of moving to GWT 2.8.0 which will allow usage of Java8 API which contains a powerful function/consumer support. So it is just a temporary measure.